pages/dashboard/projects.vue

```diff
@@ -1,22 +1,740 @@
+      this.selectedProjects = this.projects
+    },
+    uppercaseString(str) {
+      return str.charAt(0).toUpperCase() + str.slice(1)
+    },
+    sortCurrentPage() {
+      switch (this.sortBy) {
+        case 'Name':
+          this.currentSortedTableView.sort((a, b) => {
+            if (a.title < b.title) {
+              return -1
+            }
+            if (a.title > b.title) {
+              return 1
+            }
+            return 0
+          })
+          break
+        case 'Status':
+          this.currentSortedTableView.sort((a, b) => {
+            if (a.status < b.status) {
+              return -1
+            }
+            if (a.status > b.status) {
+              return 1
+            }
+            return 0
+          })
+          break
+        case 'Role':
+          this.currentSortedTableView.sort((a, b) => {
+            if (a.role < b.role) {
+              return -1
+            }
+            if (a.role > b.role) {
+              return 1
+            }
+            return 0
+          })
+          break
+        case 'Type':
+          this.currentSortedTableView.sort((a, b) => {
+            if (a.project_type < b.project_type) {
+              return -1
+            }
+            if (a.project_type > b.project_type) {
+              return 1
+            }
+            return 0
+          })
+          break
+        default:
+          break
+      }
+    },
+    switchPage(index) {
+      this.currentSortedTableView = this.pages[index - 1]
+      this.sortCurrentPage()
+      this.$nextTick()
+    },
+    changeAmountPerPage() {
+      this.pageCount = Math.ceil(this.projects.length / this.maxPerPage)
+      for (let index = 0; index < this.pageCount; index++) {
+        this.pages[index] = []
+      }
+      let counter = 0
+      let index = 0
+      this.projects.forEach((project) => {
+        if (counter > this.maxPerPage) {
+          counter = 0
+          index++
+        }
+
+        this.pages[index].push(project)
+      })
+      this.switchPage(1)
+      this.sortCurrentPage()
+    },
+    bulkDeleteSelected() {
+      try {
+        this.selectedProjects
+          .filter(
+            (it) =>
+              (it.permissions & this.DELETE_PROJECT) !== this.DELETE_PROJECT
+          )
+          .forEach(async (project) => {
+            await this.$axios.delete(
+              `project/${project.id}`,
+              this.$defaultHeaders()
+            )
+            await this.$store.dispatch('user/fetchProjects')
+            await this.$router.push(`/user/${this.$auth.user.username}`)
+            this.$notify({
+              group: 'main',
+              title: 'Action Success',
+              text: 'Project(s) has been successfully deleted.',
+              type: 'success',
+            })
+          })
+      } catch (e) {
+        this.$notify({
+          group: 'main',
+          title: 'An error occurred',
+          text: e,
+          type: 'error',
+        })
```

Comment:
This error handling system is somewhat flawed because it will show an error if a single project fails to delete, even if the other ones succeed.